harvester/utils.py

```diff
@@ -119,11 +113,39 @@ def calculate_batch_size(col_size):
         return 500
 
 
-def split_into_batches(bindings):
+def split_into_download_batches(bindings):
     """
     Split a collection into download batches.
     """
     col_size = len(bindings)
     batch_size = calculate_batch_size(col_size)
     batches = [bindings[i : i + batch_size] for i in range(0, col_size, batch_size)]
     return batches
+
+
+def bindings_with_prefix(bindings, prefix):
+    """
+    Find DC identifiers of which binding ID start with a given prefix.
+    """
+    return [
+        binding
+        for binding in bindings
+        if binding_id_from_dc(binding).startswith(prefix)
+    ]
+
+
+def assign_bindings_to_images(bindings, max_bindings_per_image, shared_prefix=""):
+    """
+    Split a list of bindings into images, each image containing no more than
+    max_bindings_per_image bindings.
+    """
+    if len(bindings) <= max_bindings_per_image:
+        return [{"prefix": shared_prefix, "bindings": bindings}]
+    images = []
+    for i in range(10):
+        prefix = shared_prefix + str(i)
```

Comment:
The other option would be that we create empty images, right? I guess in the future if we encounter new bindings with a prefix that does not match to an existing image, we create a new image with the same logic as the existing ones.

Reply:
The logic will be a bit complicated: if you see that 123/ doesn't exist, you can't assume that either 12/, 1/ or / is the correct image, because there might exist any of 121/, 122/, 124/, etc.

Buut I guess that's not such a big deal. If @aajarven also says no problem, this looks good to me!

Reply:
This solution does indeed return image 0 (with zero bindings inside), as well as other empty images with siblings that do contain bindings. For example dividing collection collection-361 with max 200 bindings/image (nb: totally toy example, in reality this is such a small collection that we would likely have it in one image) produces the following `prefix: n_bindings` pairs:
```
0:	0
1:	139
20:	105
21:	82
22:	19
23:	0
24:	0
25:	0
26:	0
27:	0
28:	0
29:	0
3:	0
4:	0
5:	0
6:	0
7:	0
8:	0
9:	0
```
What we do with these empty bindings is then left as an exercise for the programmer when we get to KP-7020. In many ways it would simplify life for both us and our users if we created the empty images, but in terms of mount points that would be somewhat wasteful. At least the zero-image can and should be omitted, but whether it should be done here or by the caller depends on what the caller will end up looking like.

Not creating the empty images does indeed make the incremental addition to pre-existing images more difficult, as we need to check what is the longest prefix length at which the newly-added binding still has siblings. Slightly more complex and a bit more error-prone, but I don't see why that would not be doable.

But all in all, unless we already know what we want, I'd leave the tweaking for later, as doing anything too specific now runs a high risk of optimizing for something that will not be our actual solution.

Reply:
Good points from both of you! We need to keep all this in mind in KP-7020, but I would agree that since we don't have a clear idea about what want, we should leave this for the next ticket and preferably have some discussions together thinking of the pros and cons of different approaches.